### PR TITLE
[tensorflow][ei][build] Update EIA Tensorflow to 2.3

### PR DESCRIPTION
**Pull request by andrewjguo**
*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [x] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [x] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [x] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
Added the Dockerfile for EIA TensorFlow 2.3, and updated the buildspec-eia.yml to build EIA TF 2.3 by default.

*Tests run:*

*DLC image/dockerfile:*

*Additional context:*
This PR won't build for now, since the EIA TensorFlow 2.3 binary hasn't been released to public s3 bucket. I have tested with the binary in beta stage locally, and will update this PR once it's promoted to public s3 bucket. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/conftest.py [hunk 1]
```diff
@@ -56,6 +56,7 @@
 
 def pytest_addoption(parser):
     parser.addoption('--region', default='us-west-2')
+    parser.addoption('--framework-version', default='')
```
Comment (saimidu) [inline test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/conftest.py:59]: The `versions` input is meant to handle incoming framework versions.

File: tensorflow/inference/docker/2.3.0/py36/Dockerfile.eia [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+LABEL maintainer="Amazon AI"
+# Specify LABEL for inference pipelines to use SAGEMAKER_BIND_TO_PORT
+# https://docs.aws.amazon.com/sagemaker/latest/dg/inference-pipeline-real-time.html
+LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
+LABEL dlc_major_version="1"
+
+# Add arguments to achieve the version, python and url
+ARG PYTHON=python3
+ARG PYTHON_VERSION=3.6.13
```
Comment (saimidu) [inline tensorflow/inference/docker/2.3.0/py36/Dockerfile.eia:11]: Please verify that this dockerfile should build for Python 3.6, and not Python 3.7.

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/conftest.py (hunk 1 above)
Comment (andrewjguo) [reply test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/conftest.py:59]: I was having a test failure for not having this field back in Jan, will remove it from this PR.

File: tensorflow/inference/docker/2.3.0/py37/Dockerfile.eia [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    cython==0.29.21 \
+    falcon==2.0.0 \
+    gunicorn==20.0.4 \
+    gevent==1.5.0 \
+    requests==2.24.0 \
+    grpcio==1.31.0 \
+    protobuf==3.12.4 \
+# using --no-dependencies to avoid installing tensorflow binary
+ && ${PIP} install --no-dependencies --no-cache-dir \
+    tensorflow-serving-api==2.3.0
```
Comment (saimidu) [inline tensorflow/inference/docker/2.3.0/py37/Dockerfile.eia:88]: Please pin to `2.3` rather than `2.3.0`.

**Review by saimidu — COMMENTED** [review]
APPROVED. Please revert temp changes, and address the one comment.

**Review by saimidu — APPROVED** [review]

**Pull request merged**